File: common/src/main/java/com/jmsoftware/maf/common/domain/authcenter/user/LoginPayload.java
package com.jmsoftware.maf.common.domain.authcenter.user;

import lombok.Data;
import org.hibernate.validator.constraints.Length;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

/**
 * Description: LoginPayload, change description here.
 *
 * @author 钟俊（zhongjun）, email: zhongjun@toguide.cn, date: 12/22/2020 6:26 PM
 **/
@Data
public class LoginPayload {
    /**
     * The Login token: username / email
     */
    @NotEmpty
    @Length(max = 100)
    private String loginToken;
    /**
     * The Password.
     */
    @NotEmpty
    @Length(max = 60)
    private String password;
    /**
     * Remember me
     */
    @NotNull
    private Boolean rememberMe;
}

File: common/src/main/kotlin/com/jmsoftware/maf/common/domain/authcenter/user/LoginPayload.kt
package com.jmsoftware.maf.common.domain.authcenter.user

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import org.hibernate.validator.constraints.Length

/**
 * Description: LoginPayload, change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/16/22 8:08 PM
 */
class LoginPayload {
    /**
     * The Login token: username / email
     */
    @NotBlank
    @Length(max = 100)
    lateinit var loginToken: String

    /**
     * The Password.
     */
    @NotBlank
    @Length(max = 60)
    lateinit var password: String

    /**
     * Remember me
     */
    @NotNull
    var rememberMe: Boolean = false
}
